### PR TITLE
Snowflake creation flow: step 1 basic form + post credentials to oauth

### PR DESCRIPTION
**Pull request by PopDaph**
## Description

First PR of the Snowflake connector creation flow. In the end, the flow will be: 
1/  User fills the form with credentials. 
2/ We post credentials to OAuth service and get the cred_id.
3/ We post to create the connector with cred_id: connector is checking the connection to see if valid, if not it displays the error on the form without creating the connector. If valid we create the connector. 
4/ If connector was created, we display the permission tree to select nodes.  

On this PR I focus on 1/ and 2/ : 
- Introduce a new modal with a specific form for the setup of Snowflake. The form validation is quite basic at the moment, we only check that all fields are set. 
- Introduce a new route with associated business logic to post credentials from front (`/w/[wId]/credentials` with expected role admin). 
- Form posts the credentials and displays an alert if successful.

<kbd>
Screenshot 2024-09-26 at 11 42 38
</kbd>

## Risk

This flow is only available if the feature flag for snowflake is available. 

## Deploy Plan

Deploy front.

### CODE REVIEW

File: front/components/data_source/CreateConnectionSnowflakeModal.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+              />
+            </div>
+
+            <div className="flex justify-center pt-2">
+              <Button.List isWrapping={true}>
+                <Button
+                  variant="primary"
+                  size="md"
+                  icon={CloudArrowLeftRightIcon}
+                  onClick={async () => {
```
Comment (fontanierh) [inline front/components/data_source/CreateConnectionSnowflakeModal.tsx:178]: nit: the async / await here is not doing anything (onClick itself is not awaited)

File: front/components/vaults/AddConnectionMenu.tsx [hunk 2]
```diff
@@ -222,29 +226,43 @@ export const AddConnectionMenu = ({
...
+                CONNECTOR_CONFIGURATIONS[connectorProvider]
+              }
+              isOpen={isOpen}
+              onClose={() =>
+                setShowConfirmConnection((prev) => ({
+                  isOpen: false,
+                  integration: prev.integration,
+                }))
+              }
+              onConfirm={async () => {
```
Comment (fontanierh) [inline front/components/vaults/AddConnectionMenu.tsx:256]: same here for the async await

**Review by fontanierh — APPROVED** [review]
exciting 🔥

**Comment by Fraggle** [comment]
> 3/ We post to create the connector with cred_id: connector is checking the connection to see if valid, if not it displays the error on the form without creating the connector. If valid we create the connector.

I actually wanted to talk about this flow. Don't you think we should be able to delete the credentials if invalid ?

**Comment by fontanierh** [comment]
> I actually wanted to talk about this flow. Don't you think we should be able to delete the credentials if invalid ?

agree it's something we should think about, but not an immediate concern / not critical for release.
I guess we could have `front` call a delete endpoint on oauth if needed

**Comment by Fraggle** [comment]
> I guess we could have front call a delete endpoint on oauth if needed

Adding a card for the delete endpoint (agree: not urgent)

**Pull request merged**